app/views/compute_resources_vms/form/azurerm/_base.html.erb

```diff
@@ -77,121 +77,114 @@
-                                                          :spinner_id    => 'load_subnets_indicator',
-                                                          :class         => 'btn-success',
-                                                          :spinner_class => 'spinner-inverse'
-                                                      })
-                 }
+<% #This view has been modified and wraps the properties of azure_vm
+   # using fields_for rails method.
 %>
 
-<%= selectable_f f, :resource_group, resource_groups,
+<%= f.fields_for :azure_vm do |vm_f| %>
+
+  <%= selectable_f vm_f, :location, compute_resource.regions,
+               { :include_blank => _('Please select an Azure region') },
+               {
+                   :label       => _('Azure Region'),
+                   :required    => true,
+                   :id          => 'azure_rm_region',
+                   :label_size  => "col-md-2",
+                   :onchange    => 'azure_rm_region_callback();',
+                   :help_inline => spinner_button_f(f, _('Reload Images, Sizes, vNets'),
+                                                    'azure_rm_region_callback();',
+                                                    {
+                                                        :id            => 'load_subnets_btn',
+                                                        :spinner_id    => 'load_subnets_indicator',
+                                                        :class         => 'btn-success',
+                                                        :spinner_class => 'spinner-inverse'
+                                                    })
+               }
+  %>
+
+  <%= selectable_f vm_f, :resource_group, resource_groups,
                  { :include_blank => true },
                  {
                      :disabled => resource_groups.empty?,
                      :label    => _('Resource Group'),
                      :required => true,
                      :id       => 'azure_rm_rg'
                  }
-%>
+  %>
 
-<%= selectable_f f, :vm_size, [],
+  <%= selectable_f vm_f, :vm_size, [],
                  { :include_blank => _('Please first select an Azure region') },
                  {
                      :label    => _('VM Size'),
                      :required => true,
                      :id       => 'azure_rm_size'
                  }
-%>
+  %>
 
-<%= selectable_f f, :platform, %w(Linux Windows),
+  <%= selectable_f vm_f, :platform, %w(Linux Windows),
                  {},
                  {
                      :label    => _('Platform'),
                      :required => true
                  }
-%>
+  %>
 
-<%= text_f f, :username,
+  <%= text_f vm_f, :username,
            {
                :label    => _('Username'),
                :required => true
            }
-%>
-<%= text_f f, :password,
+  %>
+
+  <%= text_f vm_f, :password,
```

Comment:
@ShimShtein It's a bit tricky here that password field is of type text. When it is made `password_f`, it doesn't work. I tried earlier assigning the `placeholder` option with the default values as `********` as it is defined in Foreman core, but I'm not sure if that was the go to way.